### ISSUE TITLE
fatal: reference is not a tree: 3c7c22bfbxxxxxxxxxxxxxxxxxxxxxxxxxxxxx

### ISSUE THREAD

**Issue by alessiomagribvt** [allegroai/clearml-server#205]
Good evening everyone,

working with clearml-agent I found a strange error that blocked me while I was trying to run a cloned task.
I created a new task and than cloned it from webapp, I initialized the clearml-agent (with copy and paste of credentials from webapp-popup) and finally I launched "clearml-agent daemon". 
The cloned task switched from "Running" to "Failed" state. 
Console section show me this error (sorry for some information blurred): 

```
From https://OurGitLab/OurRepo
 * branch            HEAD       -> FETCH_HEAD
fatal: reference is not a tree: 3c7c22bfbxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
Repository cloning failed: Command '['git', 'checkout', '3c7c22bfbxxxxxxxxxxxxxxxxxxxxxxxxxxxxx', '--force']' returned non-zero exit status 128.
clearml_agent: ERROR: Failed cloning repository. 
1) Make sure you pushed the requested commit:
(repository='https://OurGitLab/OurRepo', branch='clearml', commit_id='3c7c22bfbxxxxxxxxxxxxxxxxxxxxxxxxxxxxx', tag='', docker_cmd=None, entry_point='myscript.py', working_dir='.')
2) Check if remote-worker has valid credentials [see worker configuration file]
2023-07-06 16:39:11
Process failed, exit code 
```

Thinking it was a problem of our GitLab I tried to move my script inside a GitHub repository, but nothing changed. 
Thanking you in advance for your cooperation.

**Comment by jkhenning** [comment]
Hi @alessiomagribvt, can you please attached the complete agent log? Also, what cleamrl-agent version are you using?

**Comment by alessiomagribvt** [comment]
Hi @jkhenning following the complete log from the webapp -> task  -> console section:

`1688997298425 DevAlessio:0 INFO task ca969d9972b846c1a7914db57d881104 pulled from 768bdd28679941c39c2b4b150548c329 by worker DevAlessio:0

1688997313452 DevAlessio:0 DEBUG Current configuration (clearml_agent v1.5.2, location: /tmp/.clearml_agent.aelkmvg_.cfg):
----------------------
sdk.storage.cache.default_base_dir = ~/.clearml/cache
sdk.storage.cache.size.min_free_bytes = 10GB
sdk.storage.direct_access.0.url = file://*
sdk.metrics.file_history_size = 100
sdk.metrics.matplotlib_untitled_history_size = 100
sdk.metrics.images.format = JPEG
sdk.metrics.images.quality = 87
sdk.metrics.images.subsampling = 0
sdk.metrics.tensorboard_single_series_per_graph = false
sdk.network.metrics.file_upload_threads = 4
sdk.network.metrics.file_upload_starvation_warning_sec = 120
sdk.network.iteration.max_retries_on_server_error = 5
sdk.network.iteration.retry_backoff_factor_sec = 10
sdk.aws.s3.key = 
sdk.aws.s3.region = 
sdk.aws.boto3.pool_connections = 512
sdk.aws.boto3.max_multipart_concurrency = 16
sdk.log.null_log_propagate = false
sdk.log.task_log_buffer_capacity = 66
sdk.log.disable_urllib3_info = true
sdk.development.task_reuse_time_window_in_hours = 72.0
sdk.development.vcs_repo_detect_async = true
sdk.development.store_uncommitted_code_diff = true
sdk.development.support_stopping = true
sdk.development.default_output_uri = 
sdk.development.force_analyze_entire_repo = false
sdk.development.suppress_update_message = false
sdk.development.detect_with_pip_freeze = false
sdk.development.worker.report_period_sec = 2
sdk.development.worker.ping_period_sec = 30
sdk.development.worker.log_stdout = true
sdk.development.worker.report_global_mem_used = false
api.version = 1.5
api.verify_certificate = true
api.default_version = 1.5
api.http.max_req_size = 15728640
api.http.retries.total = 240
api.http.retries.connect = 240
api.http.retries.read = 240
api.http.retries.redirect = 240
api.http.retries.status = 240
api.http.retries.backoff_factor = 1.0
api.http.retries.backoff_max = 120.0
api.http.wait_on_maintenance_forever = true
api.http.pool_maxsize = 512
api.http.pool_connections = 512
api.api_server = http://devalessio.pdr.net:8008
api.web_server = http://devalessio.pdr.net:8080
api.files_server = http://devalessio.pdr.net:8081
api.credentials.access_key = 40W4S8LOUB7RJNZWXCX2
api.host = http://devalessio.pdr.net:8008
agent.worker_id = DevAlessio:0
agent.worker_name = DevAlessio
agent.force_git_ssh_protocol = false
agent.python_binary = 
agent.package_manager.type = pip
agent.package_manager.pip_version.0 = <20.2 ; python_version < '3.10'
agent.package_manager.pip_version.1 = <22.3 ; python_version >\= '3.10'
agent.package_manager.system_site_packages = false
agent.package_manager.force_upgrade = false
agent.package_manager.conda_channels.0 = pytorch
agent.package_manager.conda_channels.1 = conda-forge
agent.package_manager.conda_channels.2 = defaults
agent.package_manager.priority_optional_packages.0 = pygobject
agent.package_manager.torch_nightly = false
agent.package_manager.poetry_files_from_repo_working_dir = false
agent.venvs_dir = /home/master/.clearml/venvs-builds
agent.venvs_cache.max_entries = 10
agent.venvs_cache.free_space_threshold_gb = 2.0
agent.venvs_cache.path = ~/.clearml/venvs-cache
agent.vcs_cache.enabled = true
agent.vcs_cache.path = /home/master/.clearml/vcs-cache
agent.venv_update.enabled = false
agent.pip_download_cache.enabled = true
agent.pip_download_cache.path = /home/master/.clearml/pip-download-cache
agent.translate_ssh = true
agent.reload_config = false
agent.docker_pip_cache = /home/master/.clearml/pip-cache
agent.docker_apt_cache = /home/master/.clearml/apt-cache
agent.docker_force_pull = false
agent.default_docker.image = nvidia/cuda:10.2-cudnn7-runtime-ubuntu18.04
agent.enable_task_env = false
agent.hide_docker_command_env_vars.enabled = true
agent.hide_docker_command_env_vars.parse_embedded_urls = true
agent.abort_callback_max_timeout = 1800
agent.docker_internal_mounts.sdk_cache = /clearml_agent_cache
agent.docker_internal_mounts.apt_cache = /var/cache/apt/archives
agent.docker_internal_mounts.ssh_folder = ~/.ssh
agent.docker_internal_mounts.ssh_ro_folder = /.ssh
agent.docker_internal_mounts.pip_cache = /root/.cache/pip
agent.docker_internal_mounts.poetry_cache = /root/.cache/pypoetry
agent.docker_internal_mounts.vcs_cache = /root/.clearml/vcs-cache
agent.docker_internal_mounts.venv_build = ~/.clearml/venvs-builds
agent.docker_internal_mounts.pip_download = /root/.clearml/pip-download-cache
agent.apply_environment = true
agent.apply_files = true
agent.custom_build_script = 
agent.disable_task_docker_override = false
agent.git_user = a.magri
agent.default_python = 3.10
agent.cuda_version = 0
agent.cudnn_version = 0

Executing task id [ca969d9972b846c1a7914db57d881104]:
repository = https://ourgitlab/repo
branch = clearml
version_num = 3c7c22bfbxxxxxxxxxxxxxxxxxx
tag = 
docker_cmd = 
entry_point = ML2(CLUE).py
working_dir = .

created virtual environment CPython3.10.6.final.0-64 in 221ms
  creator CPython3Posix(dest=/home/master/.clearml/venvs-builds/3.10, clear=False, no_vcs_ignore=False, global=False)
  seeder FromAppData(download=False, pip=bundle, setuptools=bundle, wheel=bundle, via=copy, app_data_dir=/home/master/.local/share/virtualenv)
    added seed packages: pip==23.1.2, setuptools==67.8.0, wheel==0.40.0
  activators BashActivator,CShellActivator,FishActivator,NushellActivator,PowerShellActivator,PythonActivator


Using cached repository in "/home/master/.clearml/vcs-cache/reponame.2d9390a18a65edf81e5293e0a404a319/ueba-research"

1688997343603 DevAlessio:0 DEBUG warning: redirecting to https://ourgitlab/repo.git/
From https://pdrgit.ourgitlab/repo
 * branch            HEAD       -> FETCH_HEAD
fatal: reference is not a tree: 3c7c22bfb2c3e98xxxxxxxxxxxxxxxxxxxx
Repository cloning failed: Command '['git', 'checkout', '3c7c22bfb2c3xxxxxxxxxxxxxxxxx', '--force']' returned non-zero exit status 128.

clearml_agent: ERROR: Failed cloning repository. 
1) Make sure you pushed the requested commit:
(repository='https://ourgitlab/repo', branch='clearml', commit_id='3c7c22bfb2c3e98xxxxxxxxxxxxxxxxxx', tag='', docker_cmd=None, entry_point='ML2(CLUE).py', working_dir='.')
2) Check if remote-worker has valid credentials [see worker configuration file]


1688997344089 DevAlessio:0 DEBUG Process failed, exit code 1
`

And than the clearml-agent version (from pip show): 

`Name: clearml-agent
Version: 1.5.2
Summary: ClearML Agent - Auto-Magical DevOps for Deep Learning
Home-page: https://github.com/allegroai/clearml-agent
Author: Allegroai
Author-email: clearml@allegro.ai
License: Apache License 2.0
Location: /home/master/anaconda3/envs/devalessio/lib/python3.9/site-packages
Requires: attrs, furl, jsonschema, pathlib2, psutil, pyjwt, pyparsing, python-dateutil, PyYAML, requests, six, urllib3, virtualenv
Required-by: `

Thank you for your answer

**Comment by alessiomagribvt** [comment]
Hi @jkhenning following the complete log from the webapp -> task -> console section:

`1688997298425 DevAlessio:0 INFO task ca969d9972b846c1a7914db57d881104 pulled from 768bdd28679941c39c2b4b150548c329 by worker DevAlessio:0

1688997313452 DevAlessio:0 DEBUG Current configuration (clearml_agent v1.5.2, location: /tmp/.clearml_agent.aelkmvg_.cfg):
sdk.storage.cache.default_base_dir = ~/.clearml/cache
sdk.storage.cache.size.min_free_bytes = 10GB
sdk.storage.direct_access.0.url = file://*
sdk.metrics.file_history_size = 100
sdk.metrics.matplotlib_untitled_history_size = 100
sdk.metrics.images.format = JPEG
sdk.metrics.images.quality = 87
sdk.metrics.images.subsampling = 0
sdk.metrics.tensorboard_single_series_per_graph = false
sdk.network.metrics.file_upload_threads = 4
sdk.network.metrics.file_upload_starvation_warning_sec = 120
sdk.network.iteration.max_retries_on_server_error = 5
sdk.network.iteration.retry_backoff_factor_sec = 10
sdk.aws.s3.key =
sdk.aws.s3.region =
sdk.aws.boto3.pool_connections = 512
sdk.aws.boto3.max_multipart_concurrency = 16
sdk.log.null_log_propagate = false
sdk.log.task_log_buffer_capacity = 66
sdk.log.disable_urllib3_info = true
sdk.development.task_reuse_time_window_in_hours = 72.0
sdk.development.vcs_repo_detect_async = true
sdk.development.store_uncommitted_code_diff = true
sdk.development.support_stopping = true
sdk.development.default_output_uri =
sdk.development.force_analyze_entire_repo = false
sdk.development.suppress_update_message = false
sdk.development.detect_with_pip_freeze = false
sdk.development.worker.report_period_sec = 2
sdk.development.worker.ping_period_sec = 30
sdk.development.worker.log_stdout = true
sdk.development.worker.report_global_mem_used = false
api.version = 1.5
api.verify_certificate = true
api.default_version = 1.5
api.http.max_req_size = 15728640
api.http.retries.total = 240
api.http.retries.connect = 240
api.http.retries.read = 240
api.http.retries.redirect = 240
api.http.retries.status = 240
api.http.retries.backoff_factor = 1.0
api.http.retries.backoff_max = 120.0
api.http.wait_on_maintenance_forever = true
api.http.pool_maxsize = 512
api.http.pool_connections = 512
api.api_server = http://devalessio.pdr.net:8008/
api.web_server = http://devalessio.pdr.net:8080/
api.files_server = http://devalessio.pdr.net:8081/
api.credentials.access_key = 40W4S8LOUB7RJNZWXCX2
api.host = http://devalessio.pdr.net:8008/
agent.worker_id = DevAlessio:0
agent.worker_name = DevAlessio
agent.force_git_ssh_protocol = false
agent.python_binary =
agent.package_manager.type = pip
agent.package_manager.pip_version.0 = <20.2 ; python_version < '3.10'
agent.package_manager.pip_version.1 = <22.3 ; python_version >= '3.10'
agent.package_manager.system_site_packages = false
agent.package_manager.force_upgrade = false
agent.package_manager.conda_channels.0 = pytorch
agent.package_manager.conda_channels.1 = conda-forge
agent.package_manager.conda_channels.2 = defaults
agent.package_manager.priority_optional_packages.0 = pygobject
agent.package_manager.torch_nightly = false
agent.package_manager.poetry_files_from_repo_working_dir = false
agent.venvs_dir = /home/master/.clearml/venvs-builds
agent.venvs_cache.max_entries = 10
agent.venvs_cache.free_space_threshold_gb = 2.0
agent.venvs_cache.path = ~/.clearml/venvs-cache
agent.vcs_cache.enabled = true
agent.vcs_cache.path = /home/master/.clearml/vcs-cache
agent.venv_update.enabled = false
agent.pip_download_cache.enabled = true
agent.pip_download_cache.path = /home/master/.clearml/pip-download-cache
agent.translate_ssh = true
agent.reload_config = false
agent.docker_pip_cache = /home/master/.clearml/pip-cache
agent.docker_apt_cache = /home/master/.clearml/apt-cache
agent.docker_force_pull = false
agent.default_docker.image = nvidia/cuda:10.2-cudnn7-runtime-ubuntu18.04
agent.enable_task_env = false
agent.hide_docker_command_env_vars.enabled = true
agent.hide_docker_command_env_vars.parse_embedded_urls = true
agent.abort_callback_max_timeout = 1800
agent.docker_internal_mounts.sdk_cache = /clearml_agent_cache
agent.docker_internal_mounts.apt_cache = /var/cache/apt/archives
agent.docker_internal_mounts.ssh_folder = ~/.ssh
agent.docker_internal_mounts.ssh_ro_folder = /.ssh
agent.docker_internal_mounts.pip_cache = /root/.cache/pip
agent.docker_internal_mounts.poetry_cache = /root/.cache/pypoetry
agent.docker_internal_mounts.vcs_cache = /root/.clearml/vcs-cache
agent.docker_internal_mounts.venv_build = ~/.clearml/venvs-builds
agent.docker_internal_mounts.pip_download = /root/.clearml/pip-download-cache
agent.apply_environment = true
agent.apply_files = true
agent.custom_build_script =
agent.disable_task_docker_override = false
agent.git_user = a.magri
agent.default_python = 3.10
agent.cuda_version = 0
agent.cudnn_version = 0

Executing task id [ca969d9972b846c1a7914db57d881104]:
repository = https://ourgitlab/repo
branch = clearml
version_num = 3c7c22bfbxxxxxxxxxxxxxxxxxx
tag =
docker_cmd =
entry_point = ML2(CLUE).py
working_dir = .

created virtual environment CPython3.10.6.final.0-64 in 221ms
creator CPython3Posix(dest=/home/master/.clearml/venvs-builds/3.10, clear=False, no_vcs_ignore=False, global=False)
seeder FromAppData(download=False, pip=bundle, setuptools=bundle, wheel=bundle, via=copy, app_data_dir=/home/master/.local/share/virtualenv)
added seed packages: pip==23.1.2, setuptools==67.8.0, wheel==0.40.0
activators BashActivator,CShellActivator,FishActivator,NushellActivator,PowerShellActivator,PythonActivator

Using cached repository in "/home/master/.clearml/vcs-cache/reponame.2d9390a18a65edf81e5293e0a404a319/ueba-research"

1688997343603 DevAlessio:0 DEBUG warning: redirecting to https://ourgitlab/repo.git/
From https://pdrgit.ourgitlab/repo

branch HEAD -> FETCH_HEAD
fatal: reference is not a tree: 3c7c22bfb2c3e98xxxxxxxxxxxxxxxxxxxx
Repository cloning failed: Command '['git', 'checkout', '3c7c22bfb2c3xxxxxxxxxxxxxxxxx', '--force']' returned non-zero exit status 128.
clearml_agent: ERROR: Failed cloning repository.

Make sure you pushed the requested commit:
(repository='https://ourgitlab/repo', branch='clearml', commit_id='3c7c22bfb2c3e98xxxxxxxxxxxxxxxxxx', tag='', docker_cmd=None, entry_point='ML2(CLUE).py', working_dir='.')
Check if remote-worker has valid credentials [see worker configuration file]
1688997344089 DevAlessio:0 DEBUG Process failed, exit code 1
`

And than the clearml-agent version (from pip show):

Name: clearml-agent Version: 1.5.2 Summary: ClearML Agent - Auto-Magical DevOps for Deep Learning Home-page: https://github.com/allegroai/clearml-agent Author: Allegroai Author-email: clearml@allegro.ai License: Apache License 2.0 Location: /home/master/anaconda3/envs/devalessio/lib/python3.9/site-packages Requires: attrs, furl, jsonschema, pathlib2, psutil, pyjwt, pyparsing, python-dateutil, PyYAML, requests, six, urllib3, virtualenv Required-by: 

Thank you for your answer

**Comment by jkhenning** [comment]
VErsion 1.5.2 should work just fine in this respect. I assume this has something to do with submodules - can you perhaps delete your VCS cache (under the `~/.clearml` folder in the machine running the agent) and try again?

**Comment by alessiomagribvt** [comment]
SOLVED ! Thank you ! Solved deleting VCS cache
